packages/editor/src/components/entities-saved-states/index.js

```diff
@@ -70,41 +66,47 @@ export default function EntitiesSavedStates( {
 				onRequestClose={ () => onRequestClose() }
 				contentLabel={ __( 'Select items to save.' ) }
 			>
-				{ Object.keys( entityRecordChangesByRecord ).map(
-					( changedKind ) =>
-						Object.keys(
-							entityRecordChangesByRecord[ changedKind ]
-						).map( ( changedName ) =>
-							Object.keys(
-								entityRecordChangesByRecord[ changedKind ][
-									changedName
-								]
-							).map( ( changedKey ) => {
-								const id = [
-									changedKind,
-									changedName,
-									changedKey,
-								];
-								return (
-									<EntitiesSavedStatesCheckbox
-										key={ id.join( ' | ' ) }
-										id={ id }
-										name={ changedName }
-										changes={
-											entityRecordChangesByRecord[
-												changedKind
-											][ changedName ][ changedKey ]
-										}
-										checked={ checkedById.get( id ) }
-										setCheckedById={ setCheckedById }
-									/>
-								);
-							} )
-						)
-				) }
+				{ dirtyEntityRecords.map( ( record ) => {
+					return (
+						<CheckboxControl
+							key={ record.key }
+							label={
+								<>
+									{ record.entity.label }
+									{ !! record.title && (
```

Comment:
Why is the title a top-level item, but the label is merged into the entity? I think both should be top-level as the label is not part of the entity as provided by the API.

Reply:
Ideally, "entity" is removed entirely from this object (it duplicates kind, name) and we should use getEntity selector. Laziness made me add it to the experimental selector.

Reply:
Yes, let's do that.